opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -408,39 +404,40 @@ private static JSONObject getCheckBoxJsonObjects(Event event, JSONObject parentO
-        if (StringUtils.isNotBlank(value)) {
-            if (AllConstants.CHECK_BOX.equals(type)) {
-                try {
-                    List<Object> vall = new ArrayList<>();
-                    if (jsonObject.has(AllConstants.OPTIONS)) {
-                        JSONArray conceptsOptions = jsonObject.getJSONArray(AllConstants.OPTIONS);
-                        for (int i = 0; i < conceptsOptions.length(); i++) {
-                            JSONObject option = conceptsOptions.getJSONObject(i);
-                            boolean optionValue = option.optBoolean(VALUE);
-                            if (optionValue) {
-                                option.put(AllConstants.TYPE, type);
-                                option.put(AllConstants.PARENT_ENTITY_ID, jsonObject.getString(OPENMRS_ENTITY_ID));
-                                option.put(KEY, jsonObject.getString(KEY));
-                                if (combineCheckboxOptionValues) {
-                                    vall.add(option.optString(AllConstants.TEXT));
-                                } else { // For options with concepts create an observation for each
-                                    createObservation(e, option, String.valueOf(option.getBoolean(VALUE)), entity);
-                                }
+        String entity =  getString(jsonObject, OPENMRS_ENTITY);;
+        if (AllConstants.CHECK_BOX.equals(type)) {
+            try {
+                List<Object> optionValues = new ArrayList<>();
+                if (jsonObject.has(AllConstants.OPTIONS)) {
+                    JSONArray options = jsonObject.getJSONArray(AllConstants.OPTIONS);
+                    for (int i = 0; i < options.length(); i++) {
+                        JSONObject option = options.getJSONObject(i);
+                        boolean optionValue = option.optBoolean(VALUE);
+                        if (optionValue) {
+                            option.put(AllConstants.TYPE, type);
+                            option.put(AllConstants.PARENT_ENTITY_ID, jsonObject.getString(OPENMRS_ENTITY_ID));
+                            option.put(KEY, jsonObject.getString(KEY));
+                            if (CONCEPT.equals(entity)) {
+                                // For options with concepts create an observation for each
+                                createObservation(e, option, String.valueOf(option.getBoolean(VALUE)), entity);
+                            } else {
+                                optionValues.add(option.optString(AllConstants.TEXT));
                             }
                         }
-                        if (combineCheckboxOptionValues) { // For options without concepts combine the values into one observation
-                            createObservation(e, jsonObject, vall);
-                        }
                     }
-                } catch (JSONException e1) {
-                    Timber.e(e1);
+                    if (!optionValues.isEmpty()) {
```

Comment:
I think that this should be mutually exclusive. Either save the list of checked options in one obs, or an obs for each selection, however not both

Reply:
That is what is happening currently (it's mutually exclusive).

However, for assured backward compatibility, it may actually make sense to have both.